packages/mdc-menu-surface/constants.ts

```diff
@@ -31,55 +30,74 @@ const cssClasses = {
 
-/** @enum {string} */
+// tslint:disable:object-literal-sort-keys
 const strings = {
   CLOSED_EVENT: 'MDCMenuSurface:closed',
   OPENED_EVENT: 'MDCMenuSurface:opened',
-  FOCUSABLE_ELEMENTS: 'button:not(:disabled), [href]:not([aria-disabled="true"]), input:not(:disabled), ' +
-  'select:not(:disabled), textarea:not(:disabled), [tabindex]:not([tabindex="-1"]):not([aria-disabled="true"])',
+  FOCUSABLE_ELEMENTS: [
+    'button:not(:disabled)', '[href]:not([aria-disabled="true"])', 'input:not(:disabled)',
+    'select:not(:disabled)', 'textarea:not(:disabled)', '[tabindex]:not([tabindex="-1"]):not([aria-disabled="true"])',
+  ].join(', '),
 };
+// tslint:enable:object-literal-sort-keys
 
-/** @enum {number} */
 const numbers = {
-  // Total duration of menu-surface open animation.
+  /** Total duration of menu-surface open animation. */
   TRANSITION_OPEN_DURATION: 120,
-  // Total duration of menu-surface close animation.
+
+  /** Total duration of menu-surface close animation. */
   TRANSITION_CLOSE_DURATION: 75,
-  // Margin left to the edge of the viewport when menu-surface is at maximum possible height.
+
+  /** Margin left to the edge of the viewport when menu-surface is at maximum possible height. */
   MARGIN_TO_EDGE: 32,
-  // Ratio of anchor width to menu-surface width for switching from corner positioning to center positioning.
+
+  /** Ratio of anchor width to menu-surface width for switching from corner positioning to center positioning. */
   ANCHOR_TO_MENU_SURFACE_WIDTH_RATIO: 0.67,
 };
 
 /**
  * Enum for bits in the {@see Corner) bitmap.
- * @enum {number}
  */
-const CornerBit = {
-  BOTTOM: 1,
-  CENTER: 2,
-  RIGHT: 4,
-  FLIP_RTL: 8,
-};
+enum CornerBit {
+  BOTTOM = 1,
+  CENTER = 2,
+  RIGHT = 4,
+  FLIP_RTL = 8,
+}
 
 /**
  * Enum for representing an element corner for positioning the menu-surface.
  *
  * The START constants map to LEFT if element directionality is left
  * to right and RIGHT if the directionality is right to left.
  * Likewise END maps to RIGHT or LEFT depending on the directionality.
- *
- * @enum {number}
  */
-const Corner = {
-  TOP_LEFT: 0,
-  TOP_RIGHT: CornerBit.RIGHT,
-  BOTTOM_LEFT: CornerBit.BOTTOM,
-  BOTTOM_RIGHT: CornerBit.BOTTOM | CornerBit.RIGHT,
-  TOP_START: CornerBit.FLIP_RTL,
-  TOP_END: CornerBit.FLIP_RTL | CornerBit.RIGHT,
-  BOTTOM_START: CornerBit.BOTTOM | CornerBit.FLIP_RTL,
-  BOTTOM_END: CornerBit.BOTTOM | CornerBit.RIGHT | CornerBit.FLIP_RTL,
-};
+enum Corner {
+  TOP_LEFT = 0,
+  TOP_RIGHT = CornerBit.RIGHT,
+  BOTTOM_LEFT = CornerBit.BOTTOM,
+  BOTTOM_RIGHT = CornerBit.BOTTOM | CornerBit.RIGHT, // tslint:disable-line:no-bitwise
+  TOP_START = CornerBit.FLIP_RTL,
+  TOP_END = CornerBit.FLIP_RTL | CornerBit.RIGHT, // tslint:disable-line:no-bitwise
+  BOTTOM_START = CornerBit.BOTTOM | CornerBit.FLIP_RTL, // tslint:disable-line:no-bitwise
+  BOTTOM_END = CornerBit.BOTTOM | CornerBit.RIGHT | CornerBit.FLIP_RTL, // tslint:disable-line:no-bitwise
+}
+
+interface MenuDimensions {
```

Comment:
Is `constants.ts` where we've been putting this sort of thing? It seems like an odd idea for interfaces to be here. Would putting them in `adapter.ts` make more sense?

Reply:
I went back and forth on that myself. @moog16 and I decided to move them from `adapter.ts` to `constants.ts` because the constants file already has enums, which feel "type-y".

We were thinking that any type whose name beings with `MDC` (i.e., index, foundation, and adapter) should have its own file; everything else would go in `constants.js`. It feels simple and straightforward to me.

We could conceivably create an entirely separate file just for interfaces and `type` definitions, but that might be overkill.

WDYT?

Reply:
IMO interfaces feel more type-y than enums (which are much more value-y than type-y), which is why I thought of `adapter.ts`. How many components have we done this with so far?

Reply:
As discussed offline, we're going to create a new `types.ts` file for interfaces and other type definitions that don't belong in `constants.ts` or `adapter.ts`.